src/components/common/CustomSelect.vue

```diff
@@ -62,6 +62,13 @@ import { UISelectOption } from "@/interfaces/UISelectOption";
                 this.options.find(
                     ({ key }: UISelectOption) => key === this.selectedKey
                 ) ?? this.options[0];
+        },
+        options() {
+            this.selected =
+                this.options.find(
+                    ({ key }: UISelectOption) => key === this.selectedKey
+                ) ?? this.options[this.options.length - 1];
+            this.$emit("update:selectedKey", this.selected?.key);
```

Comment:
So in case an option item is added, it still shows the last selected option or if not existing anymore the last item in the options array. Sounds good actually 👍